test/integration/Sandwich.t.sol

```diff
@@ -0,0 +1,221 @@
+            _attackerDepositAmount,
+            1e18,
+            hyperdrive.calculateMaxLong() / 3
+        );
+        uint256 attackerShares = depositEverlong(
+            _attackerDepositAmount,
+            attacker
+        );
+
+        // The attacker closes their short position.
+        uint256 attackerShortProceeds = closeShort(
+            attacker,
+            maturityTime,
+            _shortAmount
+        );
+
+        // The attacker redeems their Everlong shares.
+        uint256 attackerEverlongProceeds = redeemEverlong(
+            attackerShares,
+            attacker
+        );
+
+        // The bystander redeems their Everlong shares.
+        redeemEverlong(bystanderShares, bystander);
+
+        // Calculate the amount paid and the proceeds for the attacker.
+        uint256 attackerPaid = _attackerDepositAmount + attackerShortBasePaid;
+        uint256 attackerProceeds = attackerEverlongProceeds +
+            attackerShortProceeds;
+
+        // Ensure the attacker does not profit.
+        assertLt(attackerProceeds, attackerPaid);
+    }
+
+    /// @dev Tests the following scenario:
+    ///      1. Attacker adds liquidity.
+    ///      2. Bystander deposits.
+    ///      3. Attacker deposits.
+    ///      4. Attacker removes liquidity.
+    ///      5. Attacker withdraws.
+    ///      6. Bystander withdraws.
+    function sandwich_lp_instant(
+        uint256 _lpDeposit,
+        uint256 _bystanderDeposit,
+        uint256 _attackerDeposit
+    ) public {
+        // Alice is the attacker, and Bob is the bystander.
+        address attacker = alice;
+        address bystander = bob;
+
+        // The attacker adds liquidity to Hyperdrive.
+        _lpDeposit = bound(_lpDeposit, 1e18, INITIAL_CONTRIBUTION);
+        uint256 attackerLPShares = addLiquidity(attacker, _lpDeposit);
+
+        // The bystander deposits into Everlong.
+        _bystanderDeposit = bound(
+            _bystanderDeposit,
+            1e18,
+            hyperdrive.calculateMaxLong() / 3
+        );
+        uint256 bystanderEverlongShares = depositEverlong(
+            _bystanderDeposit,
+            bystander
+        );
+
+        // The attacker deposits into Everlong.
+        _attackerDeposit = bound(
+            _attackerDeposit,
+            1e18,
+            hyperdrive.calculateMaxLong() / 3
+        );
+        uint256 attackerEverlongShares = depositEverlong(
+            _attackerDeposit,
+            attacker
+        );
+
+        // The attacker removes liquidity from Hyperdrive.
+        removeLiquidity(attacker, attackerLPShares);
+
+        // The attacker redeems from Everlong.
+        uint256 attackerEverlongProceeds = redeemEverlong(
+            attackerEverlongShares,
+            attacker
+        );
+
+        everlong.rebalance();
+
+        // The bystander redeems from Everlong.
+        redeemEverlong(bystanderEverlongShares, bystander);
+
+        assertLt(attackerEverlongProceeds, _attackerDeposit);
```

Comment:
A few of these lines are missing comments. I'm not totally sure what the rebalance is doing in this case.

Reply:
Left over from experimentation, removing and commenting any other lines with missing comments :)